pkg/operator/ceph/object/user/controller.go

```diff
@@ -718,7 +718,7 @@ func (r *ReconcileObjectStoreUser) getObjectStore(storeName string) (*cephv1.Cep
 
 	for _, store := range objectStores.Items {
 		if store.Name == storeName {
-			logger.Infof("CephObjectStore %q found", storeName)
+			logger.Debugf("CephObjectStore %q found", storeName)
```

Comment:
A thought: one of the hard things to review and keep quality on is log messages to ensure all the context is present for debugging complex clusters that have multiple of many resources, and these old logs don't really give full context. 

The full context would include both the namespaced name of the object user as well as the namespaced name of the referenced object store. I would be interested to at the same time also add the additional context.

In this case, I think the change is fairly low on the helpful scale, so if you don't have time for that chang we can definitely just merge this

Reply:
Is the logic in this function correct?  Should it be using `CephObjectStoreUser.spec.clusterNamespace`?

Reply:
That field is referring to the namespace of the ceph cluster and object store, if different from the user.

Reply:
The log messages would ideally that `clusterNamespace` when the object store or ceph cluster are being referenced.

Reply:
The current code is searching for the `CephObjectStore` across all namespaces.

Reply:
I thought the cluster and the object store were always in the same namespace, and the helper `clusterStoreNamespace()` is for that purpose. Did you mean the `CephObjectStoreUser` is allowed across all namespaces?

Reply:
I think it's a viewpoint thing. From the view of the COSU, the parent COS can be in any namespace. From the view of the cluster/COS, the COSU can be in any namespace. Since this work is in the COSU controller, it seems natural to me to say that the code is searching for the COS.

The code does also list all the COSes and then iterate to find the one referenced as Josh says. I suspect that could be optimized since the name (and namespace) of the parent COS should be known in full. But maybe that's outside of the scope of updating the log messages to better identify the actors in the given context.

Reply:
I suspect that if there were multiple CephClusters, and each had its own COS with the same name, the current code would be non-deterministic as to which COS the user would be created in as there's no constraint on the namespace (of the COS).  The list operation could be converted to a namespaced get to prevent that ambiguity.

Reply:
@BlaineEXE @jhoblitt Ah that clarifies it, thanks for all the explanation. Sounds good to make that more deterministic in a follow-up PR.